torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -1041,6 +1034,9 @@ class TORCH_API ProcessGroupNCCL : public Backend {
   // timeout for the dump to finish.
   int waitTimeoutDumpInMilSec_;
 
+  // promise to coordinate flight recorder dump.
+  std::promise<bool> promiseFlightRecorderDump_;
```

Comment:
nit: maybe use a void promise if we're not consuming the value?

Reply:
oh interesting, didn't know that.